### PR TITLE
Prism forward models return nan on singular points

### LINKED ISSUE #36: Return nan when computing magnetic field inside a prism
**Description of the desired feature:**

Inline with #30, the magnetic forward modelling functions for prism should return nan on the same singular points of the gravity tensor components, but also for any point inside the prism, since the magnetic field we compute is obtained after assuming we are outside of sources.

**Are you willing to help implement and maintain this feature?** 

Sure!

### PR DESCRIPTION
**Pull request by santisoler**
Gravity and magnetic forward modelling functions for prisms return `np.nan` when the observation points falls in a singular point. These singular points vary for the field that is being computed. For the gravity tensor components, these are all the vertices and edges perpendicular to the direction(s) of the tensor component. For the magnetic fields, these are all the vertices, all the edges and all the interior points. This PR also fixes the values of the fields on the faces: diagonal gravity tensor components and magnetic field components on prism faces return the limit of the field approaching from outside. To achieve that, a 4π constant must be added when computing on the east, north or top faces. Add tests for these new features and include explanations in the docstrings.

**Relevant issues**

Fixes #36

### CODE REVIEW

File: choclo/prism/_gravity.py [hunk 1]
```diff
@@ -597,11 +608,20 @@ def gravity_ee(easting, northing, upward, prism, density):
...
-        GRAVITATIONAL_CONST
-        * density
-        * _evaluate_kernel(easting, northing, upward, prism, kernel_ee)
-    )
+    # Return nan if the observation point falls on a singular point.
+    # For the g_ee this are edges perpendicular to the easting direction
+    # (parallel to northing and upward)
+    if is_point_on_northing_edge(
+        easting, northing, upward, prism
+    ) or is_point_on_upward_edge(easting, northing, upward, prism):
```
[the remark this answers is not in the document]
Comment (santisoler) [reply choclo/prism/_gravity.py:616]: Even if I add `()` around them, Black reformats it as it is right now.

I can disable black on that if statement:
```python
    if (
        is_point_on_northing_edge(easting, northing, upward, prism)
        or is_point_on_upward_edge(easting, northing, upward, prism)
    ):  # fmt: skip
        return np.nan
```
Does it look better?

[the remark this answers is not in the document]
Comment (santisoler) [reply choclo/prism/_gravity.py]: I think it does... so I made the changes and pushed them. If you disagree with disabling Black on this, we can revert it.

[the remark this answers is not in the document]
Comment (leouieda) [reply choclo/prism/_gravity.py]: Looks good to me 👍🏽

Strange that black doesn't respect the parenthesis.

**Comment by LL-Geo** [comment]
Sorry, I missed that 😢 That looks great 👍

**Comment by santisoler** [comment]
Thanks @LL-Geo! Don't worry about the timing, we can always make changes afterwards.

**Pull request merged**